metals/src/main/scala/scala/meta/internal/metals/BloopServers.scala

```diff
@@ -276,76 +294,95 @@ final class BloopServers(
   def ensureDesiredJvmSettings(
       maybeRequestedBloopJvmProperties: Option[List[String]],
       maybeRunningBloopJvmProperties: Option[List[String]],
-      maybeJavaHome: Option[String],
+      maybeRequestedMetalsJavaHome: Option[String],
+      maybeRunningMetalsJavaHome: Option[String],
```

Comment:
Well, the way it works is this: it checks the running version from memory, then if it is None, reads from the json file! By removing them we would have to read from the json file each time by default! Is this the functionality we are looking for?

Reply:
> Well, the way it works is this: it checks the running version from memory, then if it is None, reads from the json file

I don't see where that happens, this condition:
```
 (maybeRequestedBloopJvmProperties != maybeRunningBloopJvmProperties
          && maybeRequestedBloopJvmProperties != maybeBloopGlobalJsonJvmProperties)
```
will be true at the start when both running java home and properties are None and the requested ones are Some. So we will ask to restart even if Bloop is running properly already.

It also seems we need to change both at the same time to get the pop up about restarting Bloop. It might be a bit less complex to use the global setting only instead of the one in the user configuration.

Reply:
At the start, `maybeRequestedBloopJvmProperties != maybeRunningBloopJvmProperties` will be true! Then `maybeRequestedBloopJvmProperties != maybeBloopGlobalJsonJvmProperties` will be false! This means that 
`(maybeRequestedBloopJvmProperties != maybeRunningBloopJvmProperties
          && maybeRequestedBloopJvmProperties != maybeBloopGlobalJsonJvmProperties)` will be false! 
Therefor, at the start we are not going to ask to restart Bloop!

Reply:
> It also seems we need to change both at the same time to get the pop up about restarting Bloop
  No, how did you deduct that?
according to this condition:

 if (maybeRequestedBloopJvmProperties != maybeRunningBloopJvmProperties
           && maybeRequestedBloopJvmProperties != maybeBloopGlobalJsonJvmProperties) ||
           (maybeRequestedMetalsJavaHome != maybeRunningMetalsJavaHome
             && maybeRequestedMetalsJavaHome != maybeBloopGlobalJsonJavaHome)
changing just one of either `bloopJvmProperties` or `javaHome` triggers the update! Because they are connected by `||`

Reply:
> Well, the way it works is this: it checks the running version from memory, then if it is None, reads from the json file!

I guess the problem is that we are always reading from the bloop json file anyway! So I think I should just remove the `running` method arguments or change the semantics to the quoted logic?

Reply:
I can right it as just 
```
 if maybeRequestedBloopJvmProperties != maybeBloopGlobalJsonJvmProperties ||
            maybeRequestedMetalsJavaHome != maybeBloopGlobalJsonJavaHome
```
for simplicity!

Reply:
Ach, you are right I misread the conditions, which is a perfect example of it being a bit complicated :sweat: 

Maybe we don't need the running since each time we change to requested that gets saved to global config, which should be a good source of truth.

Reply:
Yep, I think it should be much simpler this way.

Reply:
Ok, no problem!